libs/use-publisher/src/index.ts

```diff
@@ -192,16 +202,55 @@ const usePublisher = ({
       return;
     }
 
+    // Validate source id for duplicates
+    if (broadcastOptions.sourceId) {
+      const allSourceIds = Array.from(sources).map(([, { broadcastOptions }]) => broadcastOptions.sourceId);
+
+      const dedupedSourceId = validateDuplicateSourceIds(broadcastOptions.sourceId, allSourceIds);
```

Comment:
`validate` means you are checking if it's valid, by default it should return true or false. I guess you want to modify the source id and make it unique, isn't it?